compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +289,118 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
   if (search != notedLocations_.end()) {
     assert(!search->second.isEmpty());
     idToLocation_[ast->id()] = search->second;
+    if (ieNode) {
+      checkConfigPreviouslyUsed(ast, configNameUsed);
+    }
   } else {
     assert(false && "Location for all ast should be set by noteLocation");
   }
 }
 
-AstList Builder::flattenTopLevelBlocks(AstList lst) {
+  void Builder::checkConfigPreviouslyUsed(const AstNode* ast, std::string& configNameUsed) {
+    parsing::useConfigParam(context(), configNameUsed, ast->id());
+    auto usedId = parsing::nameToConfigParamId(context(), configNameUsed);
+
+    if (usedId != ast->id()) {
+      std::string err = "ambiguous config name (";
+      err += configNameUsed;
+      err += ")";
+      std::string otherLoc = "also defined here";
+      std::string disambiguate = "(disambiguate using -s<modulename>.";
+      disambiguate += configNameUsed;
+      disambiguate += "...)";
+      ErrorMessage errorMessage = ErrorMessage(ast->id(), idToLocation_[ast->id()], err, ErrorMessage::ERROR );
+      ErrorMessage noteOtherLoc = ErrorMessage(ast->id(), idToLocation_[usedId], otherLoc, ErrorMessage::NOTE);
+      ErrorMessage noteDisambiguate = ErrorMessage(ast->id(), idToLocation_[usedId], disambiguate, ErrorMessage::NOTE);
+      addError(errorMessage);
+      addError(noteOtherLoc);
+      addError(noteDisambiguate);
+    }
+  }
+
+  std::tuple<AstNode*,std::string> Builder::checkAndUpdateConfig(AstNode* ast, pathVecT& pathVec) {
+    AstNode* ret = nullptr;
+    std::string configUsed;
+    auto configs = parsing::configParams(this->context());
```

Comment:
This could require some work so can we compute it only if we know it's a config variable? (I.e. move this variable inside of the `if (var->isConfig())` below?

Also I can never remember C++ value vs ref rules with auto. Could you put `const auto& configs` to make it clear we are not copying it?